query-algebrizer/src/clauses/where_fn.rs

```diff
@@ -243,6 +252,172 @@ impl ConjoiningClauses {
 
         Ok(())
     }
+
+
+    /// This function:
+    /// - Resolves variables and converts types to those more amenable to SQL.
+    /// - Ensures that the predicate functions name a known operator.
+    /// - Accumulates a `NumericInequality` constraint into the `wheres` list.
+    #[allow(unused_variables)]
+    pub fn apply_fulltext<'s>(&mut self, schema: &'s Schema, where_fn: WhereFn) -> Result<()> {
+        if where_fn.args.len() != 3 {
+            bail!(ErrorKind::InvalidNumberOfArguments(where_fn.operator.clone(), where_fn.args.len(), 3));
+        }
+
+        if where_fn.binding.is_empty() {
+            // The binding must introduce at least one bound variable.
+            bail!(ErrorKind::InvalidBinding(where_fn.operator.clone(), BindingError::NoBoundVariable));
+        }
+
+        if !where_fn.binding.is_valid() {
+            // The binding must not duplicate bound variables.
+            bail!(ErrorKind::InvalidBinding(where_fn.operator.clone(), BindingError::RepeatedBoundVariable));
+        }
+
+        for var in &where_fn.binding.variables() {
+            if let &Some(ref var) = var {
+                if self.input_variables.contains(var) {
+                    bail!(ErrorKind::InvalidBinding(where_fn.operator.clone(), BindingError::BoundInputVariable));
+                }
+            }
+        }
+
+        let bindings = match where_fn.binding {
+            Binding::BindRel(bindings) => {
+                if bindings.len() != 4 {
```

Comment:
In practice, Datomic uses implicit blanks for bindings, not just data patterns. Example:

http://stackoverflow.com/a/24753310/22003

I don't think we have enough specificity in the parser to flesh these out there, so we should allow it here.